cpp/src/io/csv/csv_reader.cu

```diff
@@ -515,35 +517,35 @@ gdf_error read_csv(csv_read_arg *args)
 		if (fstat(fd, &st)) { close(fd); checkError(GDF_FILE_ERROR, "cannot stat file");   }
 	
 		const auto file_size = st.st_size;
-		const auto page_size = sysconf(_SC_PAGESIZE);
+		// Can't map an empty file, will return an empty dataframe further down
+		if (file_size != 0) {
+			if (args->byte_range_offset >= (size_t)file_size) {
```

Comment:
Should we add a check somewhere that `args->byte_range_offset` is zero when `file_size` is zero?  Otherwise we could end up with an invalid, non-zero pointer `h_uncomp_data` when `compression_type == "none"`.  See line 563.

That pointer should never be de-referenced but you never now what happens in the future when someone else makes further changes.

Reply:
Modified the check to allow zero size input caused by byte range offset (when offset == size). Since we now allow zero size input, I think this version of the check is consistent.